crates/libwebrtc-sys/src/lib.rs

```diff
@@ -1888,3 +1912,863 @@ impl VideoSinkInterface {
+                    )?;
+                RtcStatsType::RtcRemoteInboundRtpStreamStats {
+                    local_id: stats.local_id.take(),
+                    round_trip_time: stats.round_trip_time.take(),
+                    fraction_lost: stats.fraction_lost.take(),
+                    round_trip_time_measurements: stats
+                        .round_trip_time_measurements
+                        .take(),
+                }
+            }
+            T::RTCRemoteOutboundRtpStreamStats => {
+                let mut stats =
+                    webrtc::cast_to_rtc_remote_outbound_rtp_stream_stats(
+                        stats.stats,
+                    )?;
+                RtcStatsType::RtcRemoteOutboundRtpStreamStats {
+                    local_id: stats.local_id.take(),
+                    remote_timestamp: stats.remote_timestamp.take(),
+                    reports_sent: stats.reports_sent.take(),
+                }
+            }
+            T::RTCTransportStats => {
+                let mut stats =
+                    webrtc::cast_to_rtc_transport_stats(stats.stats)?;
+                RtcStatsType::RtcTransportStats {
+                    packets_sent: stats.packets_sent.take(),
+                    packets_received: stats.packets_received.take(),
+                    bytes_sent: stats.bytes_sent.take(),
+                    bytes_received: stats.bytes_received.take(),
+                }
+            }
+            _ => RtcStatsType::Unimplemented,
+        };
+        Ok(res)
+    }
+}
+
+/// Represents the [stats object] constructed by inspecting a specific
+/// [monitored object].
+///
+/// [Full doc on W3C][1].
+///
+/// [stats object]: https://w3.org/TR/webrtc-stats/#dfn-stats-object
+/// [monitored object]: https://w3.org/TR/webrtc-stats/#dfn-monitored-object
+/// [1]: https://w3.org/TR/webrtc#rtcstats-dictionary
+pub struct RtcStats {
+    /// Unique ID that is associated with the object that was inspected to
+    /// produce this [RtcStats] object.
+    ///
+    /// [RtcStats]: https://w3.org/TR/webrtc#dom-rtcstats
+    pub id: String,
+
+    /// Timestamp associated with this object.
+    ///
+    /// The time is relative to the UNIX epoch (Jan 1, 1970, UTC).
+    ///
+    /// For statistics that came from a remote source (e.g., from received RtcP
+    /// packets), timestamp represents the time at which the information
+    /// arrived at the local endpoint. The remote timestamp can be found in an
+    /// additional field in an [`RtcStat`]-derived dictionary, if applicable.
+    pub timestamp_us: i64,
+
+    /// Actual stats of this [`RtcStat`].
+    ///
+    /// All possible stats are described in the [`RtcStatsType`] enum.
+    pub kind: RtcStatsType,
+}
+
+impl From<webrtc::RTCStatsWrap> for RtcStats {
+    fn from(stats: webrtc::RTCStatsWrap) -> Self {
+        let id = stats.id.clone();
+        let timestamp_us = stats.timestamp_us;
+        let kind = RtcStatsType::try_from(stats).unwrap();
```

Comment:
@rogurotus better do here `impl TryFrom` rather than `impl From` + `.unwrap()`, so there is more control outside.

Reply:
Refactored.